azavea/raster-foundry
# Add new create project modal flow

## Overview

This PR adds a multi-stage project creation modal that is initiated from the home page when the user clicks the 'Create a new Project' button. The import step of the process is stubbed, and has no effect on the project creation process. If the user chooses the 'Public Imagery' option, they are taken to the browse page.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

<img width="859" alt="screen shot 2017-03-10 at 2 11 05 pm" src="https://cloud.githubusercontent.com/assets/2442245/23810654/77262d56-05a0-11e7-9e9f-1ae91cc467cd.png">

<img width="825" alt="screen shot 2017-03-10 at 2 11 26 pm" src="https://cloud.githubusercontent.com/assets/2442245/23810667/7ea18ecc-05a0-11e7-98e4-e38fbd92528d.png">

<img width="835" alt="screen shot 2017-03-10 at 2 11 40 pm" src="https://cloud.githubusercontent.com/assets/2442245/23810672/85e19eca-05a0-11e7-8c80-9e03d21ebba1.png">

<img width="842" alt="screen shot 2017-03-10 at 2 11 48 pm" src="https://cloud.githubusercontent.com/assets/2442245/23810678/89fc5f72-05a0-11e7-853b-81f81dea2f1e.png">

<img width="837" alt="screen shot 2017-03-10 at 2 11 56 pm" src="https://cloud.githubusercontent.com/assets/2442245/23810689/92800838-05a0-11e7-90b2-355cf9f4652c.png">

### Notes

The order is a bit different (name is on the first page with the type) to make some things straightforward. Having the name on the first page allows us to create the project after the first step.

I stubbed the import page to make it functional to an extent, but left the images and side-scroller off for now until we solidify this phase's layout.


## Testing Instructions

 * Navigate to the home page `/home`
 * Click the 'Create a new Project' button
 * Test out the flow and ensure it matches the mocks (pretty closely)
 * Ensure that projects created via the flow can be found in the library

Closes #1217
Closes #1218
